Mathlib/Topology/Order/LowerUpperTopology.lean

```diff
@@ -247,6 +347,86 @@ end PartialOrder
 
 end LowerTopology
 
+
+namespace UpperTopology
+
+/-- The complements of the lower closures of finite sets are a collection of upper sets
+which form a basis for the upper topology. -/
+def upperBasis (α : Type _) [Preorder α] :=
+  { s : Set α | ∃ t : Set α, t.Finite ∧ (lowerClosure t : Set α)ᶜ = s }
+
+section Preorder
+
+variable (α)
+variable [Preorder α] [TopologicalSpace α] [UpperTopology α] {s : Set α}
+
+lemma topology_eq : ‹_› = generateFrom { s | ∃ a : α, (Iic a)ᶜ = s } := topology_eq_upperTopology
+
+variable {α}
+
+/-- If `α` is equipped with the upper topology, then it is homeomorphic to `WithUpperTopology α`.
+-/
+def withUpperTopologyHomeomorph : WithUpperTopology α ≃ₜ α :=
+  WithUpperTopology.ofUpper.toHomeomorphOfInducing ⟨by erw [topology_eq α, induced_id]; rfl⟩
+
+theorem isOpen_iff_generate_Iic_compl : IsOpen s ↔ GenerateOpen { t | ∃ a, (Iic a)ᶜ = t } s := by
+  rw [topology_eq α]; rfl
+
+instance  [Preorder α] [TopologicalSpace α] [UpperTopology α] : LowerTopology (αᵒᵈ) where
+  topology_eq_lowerTopology := by
+    refine topologicalSpace_eq ?_
+    rw [(UpperTopology.topology_eq (α))]
+
+/-- Left-infinite right-closed intervals (-∞,a] are closed in the upper topology. -/
+theorem isClosed_Iic (a : α) : IsClosed (Iic a) :=
+  isOpen_compl_iff.1 <| isOpen_iff_generate_Iic_compl.2 <| GenerateOpen.basic _ ⟨a, rfl⟩
+
+/-- The lower closure of a finite set is closed in the upper topology. -/
+theorem isClosed_lowerClosure (h : s.Finite) : IsClosed (lowerClosure s : Set α) :=
+  @LowerTopology.isClosed_upperClosure αᵒᵈ _ _ _ _ h
```

Comment:
```suggestion
  LowerTopology.isClosed_upperClosure (α := αᵒᵈ) h
```
Named arguments are really perfect in that kind of situation. Same below

Reply:
Thanks! Didn't know about that before.